matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/SyncResponsePostTreatmentAggregatorHandler.kt

```diff
@@ -93,7 +93,7 @@ internal class SyncResponsePostTreatmentAggregatorHandler @Inject constructor(
 
     private fun List<User>.saveLocally() {
         val userEntities = map { user -> UserEntityFactory.create(user) }
-        monarchy.doWithRealm {
+        monarchy.writeAsync {
```

Comment:
should this be `awaitTransaction` to force the suspend flow to be synchronous? 

I'm wondering if logic further down the flow is expecting these database values to be set

Reply:
We should have completed this before the next sync yeah

Reply:
(even if technically we can only have one transaction at a time)